app/src/parser_impl.c

```diff
@@ -0,0 +1,873 @@
+        }
+        case BIN32: {
+            return parser_msgpack_bin_type_not_supported;
+            break;
+        }
+        default: {
+            return parser_msgpack_bin_type_expected;
+            break;
+        }
+    }
+
+    if(binLen > bufferMaxSize) {
+        return parser_msgpack_bin_unexpected_size;
+    }
+
+    *bufferLen = binLen;
+    CHECK_ERROR(_readBytes(c, buff, *bufferLen))
+    return parser_ok;
+}
+
+parser_error_t _readBool(parser_context_t *c, uint8_t *value)
+{
+    uint8_t tmp = 0;
+    CHECK_ERROR(_readUInt8(c, &tmp))
+    switch (tmp)
+    {
+        case BOOL_TRUE: {
+            *value = 1;
+            break;
+        }
+
+        case BOOL_FALSE: {
+            *value = 0;
+            break;
+        }
+        default: {
+            return parser_msgpack_bool_type_expected;
+            break;
+        }
+    }
+    return parser_ok;
+}
+
+static parser_error_t _readAssetParams(parser_context_t *c, txn_asset_config *asset_config)
+{
+    size_t paramsSize = 0;
+    CHECK_ERROR(_readMapSize(c, &paramsSize))
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_MANAGER))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.manager, sizeof(asset_config->params.manager)))
+    addItem(1);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_RESERVE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.reserve, sizeof(asset_config->params.reserve)))
+    addItem(2);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_FREEZE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.freeze, sizeof(asset_config->params.freeze)))
+    addItem(3);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_CLAWBACK))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.clawback, sizeof(asset_config->params.clawback)))
+    addItem(4);
+
+    tx_num_items += 4;
+
+    // These items won't be displayed
+    if (_findKey(c, KEY_APARAMS_TOTAL) == parser_ok) {
+        CHECK_ERROR(_readInteger(c, &asset_config->params.total))
+    }
+    if (_findKey(c, KEY_APARAMS_DECIMALS) == parser_ok) {
+        CHECK_ERROR(_readInteger(c, &asset_config->params.decimals))
+    }
+    if (_findKey(c, KEY_APARAMS_DEF_FROZEN) == parser_ok) {
+        CHECK_ERROR(_readBool(c, &asset_config->params.default_frozen))
+    }
+    if (_findKey(c, KEY_APARAMS_UNIT_NAME) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.unitname, sizeof(asset_config->params.unitname)))
+    }
+    if (_findKey(c, KEY_APARAMS_ASSET_NAME) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.assetname, sizeof(asset_config->params.assetname)))
+    }
+    if (_findKey(c, KEY_APARAMS_URL) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.url, sizeof(asset_config->params.url)))
+    }
+    if (_findKey(c, KEY_APARAMS_METADATA_HASH) == parser_ok) {
+        CHECK_ERROR(_readBinFixed(c, asset_config->params.metadata_hash, sizeof(asset_config->params.metadata_hash)))
+    }
+
+    return parser_ok;
```

Comment:
is there any chance that we accept future data that is not shown at all?
Is there any version field in algorand?